packages/client/src/pages/Game/helpers.ts

```diff
@@ -0,0 +1,104 @@
+import { CANVAS_HEIGHT, CANVAS_WIDTH, MOVE_STEP } from './constants'
+import { gameState } from './gameState'
+import { TDraw } from './types'
+
+const draw: TDraw = {
+  space: ({ canvasContext }) => {
+    canvasContext.fillStyle = 'gray'
+    canvasContext.fillRect(0, 0, CANVAS_WIDTH, CANVAS_HEIGHT)
+  },
+  enemies: ({ canvasContext }) => {
+    gameState.enemies.forEach(enemy => {
+      canvasContext.fillStyle = 'lightgray'
+      canvasContext.beginPath()
+      canvasContext.arc(enemy.x, enemy.y, enemy.points, 0, 2 * Math.PI)
+      canvasContext.fill()
+
+      canvasContext.font = '16px Comic Sans MS'
+      canvasContext.fillStyle = 'red'
+      canvasContext.fillText(enemy.points.toString(), enemy.x, enemy.y)
+    })
+  },
+  hole: ({ canvasContext }) => {
+    canvasContext.fillStyle = 'black'
+    canvasContext.beginPath()
+    canvasContext.arc(
+      gameState.hole.x,
+      gameState.hole.y,
+      gameState.hole.points,
+      0,
+      2 * Math.PI
+    )
+    canvasContext.fill()
+
+    canvasContext.font = '16px Comic Sans MS'
+    canvasContext.fillStyle = 'green'
+    canvasContext.fillText(
+      gameState.hole.points.toString(),
+      gameState.hole.x,
+      gameState.hole.y
+    )
+  },
+}
+
+const swallowEnemiesNearby = () => {
+  const enemiesIndicesToSwallow: number[] = []
+
+  gameState.enemies.forEach((enemy, enemyIndex) => {
+    const distanceBetweenCentersByX = Math.abs(gameState.hole.x - enemy.x)
+    const distanceBetweenCentersByY = Math.abs(gameState.hole.y - enemy.y)
+
+    const distanceByX = Math.max(
+      distanceBetweenCentersByX - gameState.hole.points - enemy.points,
+      0
+    )
+    const distanceByY = Math.max(
+      distanceBetweenCentersByY - gameState.hole.points - enemy.points,
+      0
+    )
+
+    const haveOverlap = distanceByX === 0 && distanceByY === 0
+
+    if (haveOverlap && enemy.points < gameState.hole.points) {
+      gameState.hole.points += enemy.points
+      enemiesIndicesToSwallow.push(enemyIndex)
+    }
+
+    if (haveOverlap && enemy.points >= gameState.hole.points) {
+      gameState.hole.points -= enemy.points / 2
+      enemiesIndicesToSwallow.push(enemyIndex)
+    }
```

Comment:
Я бы всё ж рекомендовал деструктуризировать hole, как минимум hole.points:
```suggestion
    const { hole } = gameState;
    const { points: holePoints } = hole;

    const distanceBetweenCentersByX = Math.abs(hole.x - enemy.x)
    const distanceBetweenCentersByY = Math.abs(hole.y - enemy.y)

    const distanceByX = Math.max(
      distanceBetweenCentersByX - holePoints - enemy.points,
      0
    )
    const distanceByY = Math.max(
      distanceBetweenCentersByY - holePoints - enemy.points,
      0
    )

    const haveOverlap = distanceByX === 0 && distanceByY === 0

    if (haveOverlap && enemy.points < holePoints) {
      gameState.hole.points += enemy.points
      enemiesIndicesToSwallow.push(enemyIndex)
    }

    if (haveOverlap && enemy.points >= holePoints) {
      gameState.hole.points -= enemy.points / 2
      enemiesIndicesToSwallow.push(enemyIndex)
    }
```

Reply:
Спасибо, деструктуризировал `const { hole, enemies } = gameState` и стало меньше кода